app/views/comments/_comment.html.erb

```diff
@@ -5,7 +5,14 @@
     <br>
     <p>
       <strong>User:</strong>
-      <%= comment.user.name %>
+      <% if comment.anonymous && !(current_user.admin? || current_user == comment.user) %>
+        Anonymous
+      <% else %>
+        <%= comment.user.name %>
+        <% if comment.anonymous %>
+          (Posted as Anonymous)
+        <% end %>
+      <% end %>
     </p>
 
     <p>

```

Comment:
The current user won't be see their name under the comment if they're the author of the anonymous comment?